directives/submit.py

```diff
@@ -109,7 +111,18 @@ def run(self):
             u'max_group_size': data.get('max_group_size', env.config.default_max_group_size),
             u'points_to_pass': self.options.get('points-to-pass', data.get('points_to_pass', 0)),
         })
+
+        if self.content:
+            self.assert_has_content()
+            instructions = '\n'.join(self.content)
+            try:
+                parts = docutils.core.publish_parts(source=instructions, writer_name='html')
+                data.update({u'instructions': parts['fragment']})
+            except Exception as e:
+                return str(e)
```

Comment:
This seems relativle strange for exception handling. Where does the return go? Can Exception be limited to smaller scope than everything?

Reply:
I wasn't sure how to do this error handling. That return value doesn't go anywhere. Normally this submit directive doesn't do any kind of error handling. However, I am quite sure that errors of this rst->html conversion should be done here.

Reply:
Try this: `raise self.error('Error message.')`

Reply:
Like this?
`try:
  parts = ...
except Exception:
  raise self.error('Error message.')
else:
  data.update({...})`

Reply:
Yes.